packages/xo-server/src/xo-mixins/vm.js

```diff
@@ -0,0 +1,20 @@
+export default class Vms {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async revertVm({ snapshot, snapshotBefore, userId }) {
```

Comment:
We don't have in the architecture controllers which orchestrate between XO methods and XAPI methods. You can do this behavior in the API method.